### PR TITLE
chore: remove catalog_helper and use functionality in catalog_interface

### LINKED ISSUE #1003: Remove redundant catalog_helper
## Describe the bug

catalog_helper largely duplicates functionality in catalog_interface and should be removed.

### PR DESCRIPTION
**Pull request by fsuits**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary
catalog_helper duplicates functionality in catalog_interface, so it was removed
task methods that relied on catalog_helper were moved to catalog_interface
catalog_interface has the advantage of a flat dictionary of controls so that there is no need to recurse within controls for other controls.

closes #1003 

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

**Review by enikonovad — APPROVED** [review]
LGTM - looks good to me, but I would wait for @degenaro review before merging

**Review by degenaro — APPROVED** [review]
LGTM

**Pull request merged**
